### PR TITLE
vsock: expose remote podman ssh-tunnel connection to local unix socket

### PR DESCRIPTION
**Pull request by praveenkumar**
We are exposing it for podman preset but not for microshift and it is important for the tools which connect using unix socket. Also we are trying out syncing image from podman-machine instance to microshift instance using `skopeo` and that is required to make it work.

After this patch
```
$  curl --unix-socket ~/.crc/crc-http.sock http:/unix/network/services/forwarder/all | jq .
[
  {
    "local": "/home/prkumar/.crc/machines/crc/docker.sock",
    "remote": "ssh-tunnel://core@192.168.127.2:22/run/podman/podman.sock?key=%2Fhome%2Fprkumar%2F.crc%2Fmachines%2Fcrc%2Fid_ecdsa",
    "protocol": "unix"
  },
[...]
]
```

### CODE REVIEW

File: pkg/crc/machine/vsock.go [hunk 1]
```diff
@@ -93,6 +93,22 @@ func vsockPorts(preset crcPreset.Preset, ingressHTTPPort, ingressHTTPSPort uint)
 			Remote:   net.JoinHostPort(virtualMachineIP, internalSSHPort),
 		},
 	}
+
+	if preset == crcPreset.Podman || preset == crcPreset.Microshift {
```
Comment (cfergeau) [inline pkg/crc/machine/vsock.go]: We explicitly don't want this for the openshift/okd presets?
Apart from this, looks ok to me

Comment (praveenkumar) [reply pkg/crc/machine/vsock.go]: The more I think about it, looks like we can have it for each preset.

**Comment by anjannath** [comment]
not related to this PR directly..

> Also we are trying out syncing image from podman-machine instance to microshift instance using skopeo and that is required to make it work.

we can also make `podman-machine`, crc's 'podman' and 'microshift' preset, all store the images in a shared dir on the host, 
the location to use for storing the container images can be configured in the (`/etc/storage.conf`, `~/.config/containers/storage.conf`) , or we'll face some filesystem limitations because of the host<->vm shared dir?

I had tried this previously to preserve the already pulled images after `crc delete` with 'podman' preset and that seems to work mostly

**Comment by praveenkumar** [comment]
> we can also make podman-machine, crc's 'podman' and 'microshift' preset, all store the images in a shared dir on the host, the location to use for storing the container images can be configured in the (/etc/storage.conf, ~/.config/containers/storage.conf) , or we'll face some filesystem limitations because of the host<->vm shared dir? I had tried this previously to preserve the already pulled images after crc delete with 'podman' preset and that seems to work mostly

@anjannath Yes, same what I purpose but shared dir on the host is configured by user and if user choose different location then it fails, either we need to have a hard coded location which shouldn't change by user or use the podman sync mechanism. Also from podman desktop side for kind cluster they are storing the image on the disk and then loading it to the kind cluster using `kind load` command and it is easy to extend for CRC use case.

**Comment by anjannath** [comment]
> but shared dir on the host is configured by user and if user choose different location then it fails, either we need to have a hard coded location which shouldn't change by user or use the podman sync mechanism

at least for 'crc' this is hard-coded, i mean not user configurable, its always the user's full home path that is recreated in the VM

> Also from podman desktop side for kind cluster they are storing the image on the disk and then loading it to the kind cluster using kind load command and it is easy to extend for CRC use case.

i'd have to look how its handled but i guess since kind is already running on top of the `podman` in podman-machine, it might not actually be making duplicates, but for CRC we'd be duplicating the same images inside the VMs for each preset

**Pull request merged**